### PR TITLE
Fix god pot

### PR DESCRIPTION
**Pull request by heyngra**
Fixes god pot in TODO overlay
https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/432#issuecomment-1307613431

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java [hunk 1]
```diff
@@ -307,26 +307,29 @@ public void update() {
...
-						if (godpotRemaingTimeUnformatted.length >= 2) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 60 * 1000;
-							i++;
-						}
-						if (godpotRemaingTimeUnformatted.length >= 1) {
-							godPotDuration = godPotDuration + (long) Integer.parseInt(godpotRemaingTimeUnformatted[i]) * 1000;
+						long godpotRemainingTime = Integer.parseInt(activeEffectsMatcher.group(1));
+						String godpotRemainingTimeType = activeEffectsMatcher.group(2);
+						switch (godpotRemainingTimeType) {
+							case "Years":
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:315]: P sure these could be the non plural versions like the cookie "Hour" etc

**Review by NopoTheGamer — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java (hunk 1 above)
Comment (heyngra) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/TimersOverlay.java:315]: implemented it like you in https://github.com/NotEnoughUpdates/NotEnoughUpdates/pull/432/commits/99a0fe53a9d2f0973dd210d5f022642c992d9ad5

**Comment by NopoTheGamer** [comment]
Don't you also have to update the regex?

**Comment by heyngra** [comment]
oh yeah right, give me a sec

**Comment by jani270** [comment]
I would love to have a 1 year god pot

**Comment by Calvahar** [comment]
By the way, at 5 minutes 59 seconds, the text changes to:
image

**Comment by romangraef** [comment]
wow why does the god pot display 4m 17s for 5 minutes and 59 seconds, thats like a completely different thing. way to be inconsistent hypixel!

**Comment by Calvahar** [comment]
> wow why does the god pot display 4m 17s for 5 minutes and 59 seconds

Obviously exactly what I meant. I'll rephrase it for you. Starting at 5 minutes 59 seconds, the text changes to "*m *s". See image for what it looks like when there's 4 minutes 17 seconds left on the God Pot.

**Comment by heyngra** [comment]
> By the way, at 5 minutes 59 seconds, the text changes to: image

Okay, so when under 6 minutes it happens to show it in *m and *s, does anybody know if there's other weird hypickle cases like this where something magically changes to another showed units to mess up everything? 
(I'm going to try to fix this 6 minute thingie tommorow)

**Comment by heyngra** [comment]
would need someone with god pot at this "below 6 minute mark" to test if this implementation works

**Comment by hannibal002** [comment]
Why don't we grab the data from the SkyBlock menu instead and ignore the footer entirely?

image

Since users tend to open SB menus every few minutes anyway, and we have an “ate new cookie” check already. (and I want to avoid waiting 4 years to test if the sub 5 min workaround works :D)

**Comment by heyngra** [comment]
> Why don't we grab the data from the SkyBlock menu instead and ignore the footer entirely?
> 
> image
> 
> Since users tend to open SB menus every few minutes anyway, and we have an “ate new cookie” check already. (and I want to avoid waiting 4 years to test if the sub 5 min workaround works :D)

Where do you have God Potion in SB menu? Yeah, I get that you check cookie buff, but what about god potion? (and this 5m check was mostly related to god potion, even though I just realized cookie buffs probably also show these minutes and seconds at once which should also be checked)

**Comment by jani270** [comment]
> > Why don't we grab the data from the SkyBlock menu instead and ignore the footer entirely?
> > image
> > Since users tend to open SB menus every few minutes anyway, and we have an “ate new cookie” check already. (and I want to avoid waiting 4 years to test if the sub 5 min workaround works :D)
> 
> Where do you have God Potion in SB menu? Yeah, I get that you check cookie buff, but what about god potion? (and this 5m check was mostly related to god potion, even though I just realized cookie buffs probably also show these minutes and seconds at once which should also be checked)

We could check god pot with api

**Comment by NopoTheGamer** [comment]
Both sb menu and api are cringe imo. If we can constantly get the data we should do that. Also my implementation of the cookie timer should take into account minutes and seconds

**Comment by heyngra** [comment]
> Both sb menu and api are cringe imo. If we can constantly get the data we should do that. Also my implementation of the cookie timer should take into account minutes and seconds

image
pretty sure it won't without these changes (it checks only for Minute/Minutes and not m which might appear below 6 min mark, accordingly with seconds)

**Comment by NopoTheGamer** [comment]
> > Both sb menu and api are cringe imo. If we can constantly get the data we should do that. Also my implementation of the cookie timer should take into account minutes and seconds
> 
> 
> 
> image
> 
> pretty sure it won't without these changes (it checks only for Minute/Minutes and not m which might appear below 6 min mark, accordingly with seconds)

Oh maybe I should of read all the comments

**Pull request merged**